src/side-navigation/__tests__/complex-side-navigation.test.tsx

```diff
@@ -22,18 +22,26 @@ it('Side navigation with all possible items', async () => {
         type: 'link',
         text: 'Page 1',
         href: '#/page1',
+        // this test id deliberately contains double quotes
+        // to test the selector escaping
+        testId: '"link-page-1"',
```

Comment:
I'd liked to have a separate test case for this like [this PR](https://github.com/cloudscape-design/components/pull/2960/files#diff-f5d540f03d540755acdc30b25d75379f215619fb90b39d59145d41b10741a53dR101), but this particular component doesn't have any test dedicated for test utils, so to follow the existing structure I opted for this way.